### DISCUSSION TITLE
Sends wrong mqtt?

### DISCUSSION THREAD

**Discussion by Magnus-rosenborg** [arendst/Tasmota#21579] [Support request]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
My device is a sonoff touch wall switch with 3 buttons. ( T1EU3C) I have tried the template for it and also setting it by myself using the generic template. The problem is that it sends POWER1 on every button shoudn't it be POWER2 for button 2 end so on? See the image for more details. 

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ X] Device used (e.g., Sonoff Basic): _T1EU3C____
- [ X] Tasmota binary firmware version number used: _14.1.0____
  - [ ] Pre-compiled
  - [ X] Self-compiled
- [X ] Flashing tools used: _ESP-TOOL____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`: 
```lua
  Configuration output here:
12:41:07.139 MQT: stat/37/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
12:41:07.365 MQT: stat/37/RESULT = {"Module":{"0":"Generic"}}
12:41:07.621 MQT: stat/37/RESULT = {"GPIO0":{"32":"Knapp1"},"GPIO1":{"0":"Ingen"},"GPIO2":{"0":"Ingen"},"GPIO3":{"0":"Ingen"},"GPIO4":{"226":"Relä3"},"GPIO5":{"225":"Relä2"},"GPIO9":{"33":"Knapp2"},"GPIO10":{"34":"Knapp3"},"GPIO12":{"224":"Relä1"},"GPIO13":{"0":"Ingen"},"GPIO14":{"0":"Ingen"},"GPIO15":{"0":"Ingen"},"GPIO16":{"0":"Ingen"},"GPIO17":{"0":"Ingen"}}

```
- [X ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ X] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
12:42:42.483 MQT: stat/37/STATUS = {"Status":{"Module":0,"DeviceName":"HW-brytare-i-kök","FriendlyName":["HW-brytare-i-kök","HW-brytare-i-kök2","HW-brytare-i-kök3"],"Topic":"37","ButtonTopic":"KNAPP37","Power":"000","PowerLock":"000","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"SWITCH37","SwitchMode":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"ButtonRetain":1,"SwitchRetain":1,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":1}} (bevarad)
12:42:42.492 MQT: stat/37/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"","OtaUrl":"http://insidan.us.to/sonoff/firmware-mini37.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:15:39","StartupUTC":"2024-06-06T11:27:03","Sleep":50,"CfgHolder":4617,"BootCount":3,"BCResetTime":"2024-06-06T12:16:26","SaveCount":52,"SaveAddress":"F8000"}} (bevarad)
12:42:42.501 MQT: stat/37/STATUS2 = {"StatusFWR":{"Version":"14.1.0(tasmota)","BuildDateTime":"2024-06-06T13:14:14","Boot":31,"Core":"2_7_6","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"385/699"}} (bevarad)
12:42:42.509 MQT: stat/37/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["lj3m",""],"TelePeriod":900,"Resolution":"558180C0","SetOption":["000080E9","2805C8000100060000785A0A192800000000","008000C8","00006000","80004001","00000000"]}} (bevarad)
12:42:42.526 MQT: stat/37/STATUS4 = {"StatusMEM":{"ProgramSize":646,"Free":356,"Heap":23,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14325E","FlashFrequency":40,"FlashMode":"DOUT","Features":["041D","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,!3,!4,!5,!6,7,!8,9,10,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}} (bevarad)
12:42:42.536 MQT: stat/37/STATUS5 = {"StatusNET":{"Hostname":"S37","IPAddress":"192.168.200.237","Gateway":"192.168.200.254","Subnetmask":"255.255.255.0","DNSServer1":"192.168.200.254","DNSServer2":"0.0.0.0","Mac":"E8:DB:84:46:6F:51","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}} (bevarad)
12:42:42.543 MQT: stat/37/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.200.1","MqttPort":1883,"MqttClientMask":"37","MqttClient":"37","MqttUser":"scriptuser","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":60,"SOCKET_TIMEOUT":4}} (bevarad)
12:42:42.553 MQT: stat/37/STATUS7 = {"StatusTIM":{"UTC":"2024-06-06T11:42:42Z","Local":"2024-06-06T12:42:42","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"04:48","Sunset":"20:49"}} (bevarad)
12:42:42.561 MQT: stat/37/STATUS10 = {"StatusSNS":{"Time":"2024-06-06T12:42:42"}} (bevarad)
12:42:42.570 MQT: stat/37/STATUS11 = {"StatusSTS":{"Time":"2024-06-06T12:42:42","Uptime":"0T00:15:39","UptimeSec":939,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"0","POWER2":"0","POWER3":"0","Wifi":{"AP":1,"SSId":"lj3m","BSSId":"88:3D:24:C9:EB:10","Channel":1,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:07"}}} (bevarad)

```
- [ X] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:10:53.127 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:10:53.131 MQT: stat/37/RESULT = {"Button4":{"Action":"SINGLE"}}
14:10:53.275 SRC: MQTT
14:10:53.278 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
14:10:53.283 MQT: stat/37/RESULT = {"POWER1":"1"}
14:10:53.424 CFG: Sparade till flash vid F8, Räkna 156, Bytes 4096
14:10:53.691 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:10:53.698 MQT: stat/37/RESULT = {"Button4":{"Action":"CLEAR"}}
14:10:53.936 SRC: MQTT
14:10:53.939 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 0, Data '0'
14:10:53.945 MQT: stat/37/RESULT = {"POWER1":"0"}
14:10:54.374 CFG: Sparade till flash vid F7, Räkna 157, Bytes 4096
14:11:04.515 WIF: Kontrollerar anslutning...
14:11:14.851 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:11:14.858 MQT: stat/37/RESULT = {"Button5":{"Action":"SINGLE"}}
14:11:14.946 SRC: MQTT
14:11:14.949 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
14:11:14.954 MQT: stat/37/RESULT = {"POWER1":"1"}
14:11:15.405 CFG: Sparade till flash vid F6, Räkna 158, Bytes 4096
14:11:16.419 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:11:16.425 MQT: stat/37/RESULT = {"Button5":{"Action":"CLEAR"}}
14:11:16.516 SRC: MQTT
14:11:16.519 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 0, Data '0'
14:11:16.525 MQT: stat/37/RESULT = {"POWER1":"0"}
14:11:17.404 CFG: Sparade till flash vid F5, Räkna 159, Bytes 4096
14:11:20.521 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:11:20.527 MQT: stat/37/RESULT = {"Button6":{"Action":"SINGLE"}}
14:11:20.620 SRC: MQTT
14:11:20.623 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
14:11:20.629 MQT: stat/37/RESULT = {"POWER1":"1"}
14:11:21.408 CFG: Sparade till flash vid F4, Räkna 160, Bytes 4096
14:11:22.669 MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)
14:11:22.675 MQT: stat/37/RESULT = {"Button6":{"Action":"CLEAR"}}
14:11:22.767 SRC: MQTT
14:11:22.770 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 0, Data '0'
14:11:22.775 MQT: stat/37/RESULT = {"POWER1":"0"}
14:11:23.404 CFG: Sparade till flash vid FB, Räkna 161, Bytes 4096

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
I have tried to have button 1,2,3 and like 4,5,6 no difference. Even when detach buttons using SO114 still sends mqtt like power1 1 on button 4. So strange!!!  It works fine in gui. But not the touch buttons.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
if button 4 is pressed should send power4 1 or power4 0

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
![image](https://github.com/arendst/Tasmota/assets/49500329/1c64c6b0-4785-4cdf-812c-def6db3567de)
![image](https://github.com/arendst/Tasmota/assets/49500329/95e9a13c-4cd5-42ba-bc78-4ef612ab11e3)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Publishing payloads with `POWER` has nothing directly to do with what button is being pushed, but is instead about what relay is changing state. But indirectly, the default action for button/switch inputs is to toggle power.

You are not creating any relationship between buttons and relays by the number, the only thing that matters is the sequence, and indexes you assign per component type has to start at 1, and not skip any numbers.

SetOption114 (which is for Switch inputs) is not doing anything when inputs are configured as a Button, that would be SetOption73.

If you've been messing around with `ButtonTopic`, better not do that, without fully understanding this specialized command. In fact, I'd recommend going back to initial configuration, and not heaping different config options on top on each other. Your mixup of topics with KNAPP37 and just 37 also looks like your config is too much combination of attempts of doing something.

**Comment by barbudor** [comment]
MQT: cmnd/KNAPP37/POWER1 = 1 (bevarad)

If bevarad means retained, the it is a terrible idea to use retained power command
You should first of all get rid of that
